hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/artifacts/smtable/SoftwareModuleTable.java

```diff
@@ -154,14 +149,14 @@ protected boolean isMaximized() {
     }
 
     @Override
-    protected ArtifactUploadState getManagmentEntityState() {
+    protected ArtifactUploadState getManagementEntityState() {
         return artifactUploadState;
     }
 
     @Override
-    protected void publishEntityAfterValueChange(final SoftwareModule lastSoftwareModule) {
-        artifactUploadState.setSelectedBaseSoftwareModule(lastSoftwareModule);
+    protected void publishSelectedEntityEvent(final SoftwareModule lastSoftwareModule) {
         eventBus.publish(this, new SoftwareModuleEvent(BaseEntityEventType.SELECTED_ENTITY, lastSoftwareModule));
+        setLastSelectedEntity(lastSoftwareModule);
```

Comment:
This looks strange to me. Do not mix the concerns of publishing an event and set the last selected event here